File: Tagx00.Frontend/src/app/models/instance/image/job/DistrictJob.ts
import { ImageJob } from "./ImageJob";
import { Point } from "../Shapes";
import { TagDescriptionTuple } from "../../TagTuple";
import { ImageMissionType } from "../../../mission/image/ImageMission";

export interface Boundary {
  points: Point[];
}

export interface DistrictTagDescriptionTuple {
  boundaries: Boundary[];
  tagDescriptionTuple: TagDescriptionTuple;
}


export interface DistrictJob extends ImageJob {
  type: ImageMissionType.DISTRICT;
  tuples: DistrictTagDescriptionTuple[];
}

File: Tagx00.Backend/src/main/java/trapx00/tagx00/publicdatas/mission/image/district/DistrictJob.java
package trapx00.tagx00.publicdatas.mission.image.district;

import trapx00.tagx00.publicdatas.mission.image.ImageJob;
import trapx00.tagx00.vo.mission.image.ImageMissionType;

import java.util.List;

public class DistrictJob extends ImageJob {
    private List<DistrictTagDescriptionTuple> tuples;

    public DistrictJob() {
    }

    public DistrictJob(ImageMissionType type, List<DistrictTagDescriptionTuple> tuples) {
        super(type);
        this.tuples = tuples;
    }

    public List<DistrictTagDescriptionTuple> getTuples() {
        return tuples;
    }

    public void setTuples(List<DistrictTagDescriptionTuple> tuples) {
        this.tuples = tuples;
    }
}
